### PR TITLE
Move StoredValue Into `casper-types`.

### LINKED ISSUE #1823: Move `StoredValue` to casper-types.
@Fraser999 to add more details

### PR DESCRIPTION
**Pull request by nilz3ro**
# About These Changes
* Moved `StoredValue` type into `casper-types`.
* closes #1823

### CODE REVIEW

File: types/src/account.rs [hunk 1]
```diff
@@ -1,323 +1,285 @@
...
-#![allow(clippy::field_reassign_with_default)]
-
-use alloc::{format, string::String, vec::Vec};
-use core::{
-    array::TryFromSliceError,
-    convert::TryFrom,
-    fmt::{self, Debug, Display, Formatter},
-};
+mod account_hash;
+/// TODO: doc comment.
```
Comment (rafal-ch) [inline types/src/account.rs]: Could we actually add the missing doc comments at this stage?

**Review by rafal-ch — APPROVED** [review]
LGTM, but please check if these changes deserve mentioning in the changelog of `casper-types`.

**Review by Fraser999 — APPROVED** [review]

**Comment by cspramit** [comment]
@nilz3ro why #1900 referenced are the fixes linked to #1830

**Pull request merged**